src/platform/nrfconnect/BUILD.gn

```diff
@@ -39,6 +39,8 @@ static_library("nrfconnect") {
     "ConnectivityManagerImpl.h",
     "DeviceNetworkProvisioningDelegateImpl.cpp",
     "DeviceNetworkProvisioningDelegateImpl.h",
+    "DiagnosticDataProviderImpl.cpp",
```

Comment:
nRF Connect SDK is based on Zephyr so it can by default use `../Zephyr/DiagnosticDataProviderImpl.cpp` and doesn't require its own implementation.